core/src/test/java/org/apache/spark/shuffle/sort/ShuffleInMemorySorterSuite.java

```diff
@@ -75,6 +75,9 @@ public void testBasicSorting() throws Exception {
     // Write the records into the data page and store pointers into the sorter
     long position = dataPage.getBaseOffset();
     for (String str : dataToSort) {
+      if (!sorter.hasSpaceForAnotherRecord()) {
+        sorter.expandPointerArray(consumer.allocateArray(sorter.numRecords() * 2 * 2));
```

Comment:
I think this should only be `* 2` because the shuffle sorter only uses one array entry per record instead of the pair of entires which is used by the more general prefix sorter. I can fix this up myself on merge.